javascript/javascript3/week3/Shopping-cart/main.js

```diff
@@ -0,0 +1,117 @@
+const productList = document.getElementById("product-list");
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    //should add a product to the products array.
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    //should remove a product from the products array.
+    const findIndex = this.products.indexOf(product);
+    //Array.prototype.indexOf() will return -1 if the element is not found so recommend safety check around this.
+    if (findIndex > -1) {
+      this.products.splice(findIndex, 1);
+      //   product--; //will remove every matching item
+    }
+  }
+
+  searchProduct(productName) {
```

Comment:
Here could be a problem that your array will have always exactly one item, and it would need to match exactly, since you are getting the index of the product which is passed in the function. 

Instead what this function should do, is to go over the products in the shopping cart (this.products) with for example forEach and check whether the name of the searched product is the same as the name of the product in the shopping cart. if it is, push it to the array. IndexOf, would take only the first value that would exactly match. 

tip. lowercasing the search input and product names gives user a better flexibility when searching.